docs/architecture/high-level-components.md

```diff
@@ -0,0 +1,23 @@
+# Overview
+
+This document describes communication between Management Plane, Application and Runtimes. 
+
+## Components
+Management Plane is a central place which stores configuration about Applications and Runtimes and is responsible for propagating that information to Applications and Runtimes. 
+Management Plane plays crucial role in establishing trusted connection between Applications and Runtimes. 
+![](./assets/components-high-level.svg)
+
+An administrator adds Runtimes, Applications and configures them through Management Plane GraphQL API. 
+To keep Runtimes in-sync with the configuration, an agent installed on a Runtime is responsible for fetching actual configuration. 
+In case an Application has configured optional Webhooks, Management Plane notify an Application about any events that concern given Application.
+Management Plane does not participate in any business flow. After establishing a trusted connection between Application
+and Runtime, they communicate directly, with omitting Management Plane. 
+
+## Grouping
+In order to connect Applications and Runtimes, assign it to the same Group.
+A Group is a simple label with the key `group`. If Application or Runtime is not explicitly assigned to any Group, it belongs to the `default` Group.
+If you add a Group to the Application or Runtime assigned to the `default` group, it is automatically removed from the `default` group and moved to the new one.
+![](./assets/grouping.svg)
```

Comment:
Hmm, Maybe it would be good to include Runtime3, which has label `group=default`. It could be not connected with any app, but the diagram will show the concept with default group.